pkg/sigretests/main.go

```diff
@@ -0,0 +1,157 @@
+package sigretests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+const org = "kubevirt"
+const repo = "kubevirt"
+
+type failedJob struct {
+	jobName     string
+	buildNumber string
+}
+
+type SigRetests struct {
+	SigCompute  int
+	SigNetwork  int
+	SigStorage  int
+	SigOperator int
+}
+
+var redJobs []failedJob
+
+func prHistoryURL(org string, repo string, prNumber string) string {
+	return fmt.Sprintf("https://prow.ci.kubevirt.io/pr-history/?org=%s&repo=%s&pr=%s", org, repo, prNumber)
+}
+
+func prStorageURL(org string, repo string, prNumber string) string {
+	return fmt.Sprintf("https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/%s_%s/%s/", org, repo, prNumber)
+}
+
+func finishedJSONURL(org string, repo string, prNumber string, jobName string, buildNumber string) string {
+	return fmt.Sprintf("%s/%s/%s/finished.json", prStorageURL(org, repo, prNumber), jobName, buildNumber)
+}
+
+func filterFailedJobs(node *html.Node) (failedJobs []failedJob) {
+	var redJob failedJob
+	if node.Type == html.ElementNode && node.Data == "td" {
+		for _, td := range node.Attr {
+			if strings.Contains(td.Val, "run-failure") {
+				for _, href := range node.FirstChild.Attr {
+					if strings.Contains(href.Val, "e2e") {
+						buildLogUrl := strings.Split(href.Val, "/")
+						redJob.jobName = buildLogUrl[len(buildLogUrl)-2]
+						redJob.buildNumber = buildLogUrl[len(buildLogUrl)-1]
+						redJobs = append(redJobs, redJob)
+					}
+				}
+			}
+		}
+	}
+	for child := node.FirstChild; child != nil; child = child.NextSibling {
+		filterFailedJobs(child)
+	}
+	return redJobs
+}
+
+func getLatestCommit(node *html.Node) (latestCommit string) {
+	if node.Type == html.ElementNode && node.Data == "a" {
+		for _, td := range node.Attr {
+			if strings.Contains(td.Val, "commit") {
+				commitURL := strings.Split(td.Val, "/")
+				latestCommit := commitURL[len(commitURL)-1]
+				return latestCommit
+			}
+		}
+	}
+	for child := node.FirstChild; child != nil; child = child.NextSibling {
+		latestCommit = getLatestCommit(child)
+		if latestCommit != "" {
+			return latestCommit
+		}
+	}
+	return ""
+}
+
+func getFailedJobsForLatestCommit(org string, repo string, prNumber string) (failedJobs []failedJob, err error) {
+	prHistory := prHistoryURL(org, repo, prNumber)
```

Comment:
This is interesting - I have never thought about scraping the history page directly... I could imagine reusing that for metrics that we might want to shove into the prometheus.

:+1: